### ISSUE TITLE
two routers at once???

### ISSUE THREAD

**Issue by catmando** [rackt/react-router#1875]
Our help pages are presented to the user as a bootstrap modal dialog.

I am experimenting with having the router keep track of the users history while navigating sub pages within this modal dialog.

i.e. I am on a page like this:  
`cart/1234/personalize`, then I click help and get
`help/faq`                        then I might navigate to 
`help/faq/how-to-use`     and finally exit the dialog, pop the history and I am back to
`cart/1234/personalize`

Works (almost) great.  

The problem is once you bring up the modal dialog then you are no longer presenting the background page (in the above example `cart/1234/personalize`, so it basically become empty, which is disconcerting the users.  (The whole reason we like the help to be in a modal dialog is the user does not get alarmed that they might be loosing their work.)

Hopefully there is some way to stop the router from rerendering or perhaps have two routers, one which gets paused while the help page is up?  Something like that...

**Comment by catmando** [comment]
Just thinking... could I fire up a second router with a static history for url just before the help page came up?

**Comment by ryanflorence** [comment]
the methods in server rendering apply here, just use `match` and `RoutingContext` like a server, but if your'e in a component put the `renderProps` in state and pass to `RoutingContext` https://github.com/rackt/react-router/blob/master/docs/advanced/ServerRendering.md

**Comment by catmando** [comment]
hmmm... @ryanflorence I had already coded up a different approach:  I made a version of location that has can have multiple instances.  When the modal dialog comes up I push the current location state (including call backs etc) and start running with the new location state.  Each router has its own instance of location.  

Well it works "once" as in in I get the dialog up, and can route inside the dialog, back button works etc.  I can exit the dialog and every thing works fine.

However if I open the dialog a second time the route_handler generates nothing!

Could be a bug in my code, but I am wondering if there is some data in react-router that is being shared between my two router instances, that is causing the grief.

Or I also notice that that the dialog component is getting re-rendered when the main routes change, could this be messing things up?
